README.md

```diff
@@ -1,78 +1,263 @@
 # kuzzle-plugin-s3-upload
-Here, you'll find the boilerplate to start coding a new [Kuzzle Core Plugin](https://docs.kuzzle.io/plugins/1/essentials/getting-started/). A Core Plugin allows you to
 
-* [listen asynchronously](https://docs.kuzzle.io/plugins/1/hooks/), and perform operations that depend on data-related events;
-* [listen synchronously](https://docs.kuzzle.io/plugins/1/pipes/), and approve, modify and/or reject data-related queries;
-* [add a controller route](https://docs.kuzzle.io/plugins/1/controllers/) to expose new actions to the API;
-* [add an authentication strategy](https://docs.kuzzle.io/plugins/1/strategies/) to Kuzzle.
+This plugin exposes actions to upload files to Amazon S3 using [Presigned URLs](https://docs.aws.amazon.com/AmazonS3/latest/dev/PresignedUrlUploadObject.html).  
 
+## Usage
 
-The boilerplate demonstrates each feature of a Core Plugin.
+Get a Presigned URL: 
 
-## Plugin development
+```javascript
+// Kuzzle request
+{
+  "controller": "kuzzle-plugin-s3-upload/file",
+  "action": "getUploadUrl",
+  "filename": "headcrab.png",
+  "uploadDir": "xen"
+}
 
-This plugin is useful only if you use it as the starting point of your work. It's a boilerplate.
+// Kuzzle response
+{
+  "fileKey": "xen/<uuid>/headcrab.png",
+  "uploadUrl": "https://s3.eu-west-3.amazonaws.com/...",
+  "fileUrl": "https://s3.eu-west-3.amazonaws.com/..."
+}
+```
 
-### On an existing Kuzzle
+Then make a PUT request to the `uploadUrl` with the file content in the body.  
```

Comment:
```suggestion
Then send a PUT request to the `uploadUrl` URL with the body set to the file's content.
```

Also, what is the file content format? Should I send an ArrayBuffer?

Reply:
Whatever you want, text, bytes..